core-rust/state-manager/src/state_manager.rs

```diff
@@ -500,6 +480,10 @@ where
                                 invalid_at_epoch,
                                 rejection_reason: None,
                             });
+                            match success {
+                                VertexLimitsAdvanceSuccess::VertexFilled => break,
+                                VertexLimitsAdvanceSuccess::VertexNotFilled => {}
+                            }
                         }
                         Err(error) => {
                             rejected_transactions.push(RejectedTransaction {
```

Comment:
We possibly should (EG) consider stopping executing the proposal the first time a limit is breached, which may prove a more efficient use of time for the network. Otherwise a proposal full of slow transactions would still take a long time to execute, even if only (say) the first two could be committed.

Reply:
In any case, I'm looking forward to us having a node metric here so we understand how much wasted work there is executing transactions we can't fit.